### PR TITLE
feat: implement `ContributorsChart` for repo pages

### LINKED ISSUE #3293: Feature: Contributor evolution chart for repo pages
### Current behavior

N/A

### Suggested solution

image

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by zeucapua**
## Description

Adds a new chart to repo pages that shows the number of contributors in a given range, as a bar graph showing the daily distribution and overview numbers for the total and average.

## Related Tickets & Documents

Closes #3293 

## Mobile & Desktop Screenshots/Recordings
image

## Steps to QA

1. Go to repo pages and scroll down to see the Contributor chart card
2. Select different ranges to see change in numbers and graphs

## Tier (staff will fill in)

- [ ] Tier 1
- [x] Tier 2
- [ ] Tier 3
- [ ] Tier 4

## [optional] What gif best describes this PR or how it makes you feel?
cats-awesome-gif

### CODE REVIEW

**Comment by nickytonline** [comment]
The chart looks great on smaller screens. 🔥 

CleanShot 2024-05-01 at 17 05 08

Not related to this PR @isabensusan, but the charts all seem to have too much padding on the left. Having equal padding would balance things.

**Review by jpmcb — APPROVED** [review]

File: components/Graphs/ContributorsChart.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+};
+
+export default function ContributorsChart({
+  stats,
+  rangedTotal,
+  syncId,
+  range = 30,
+  isLoading,
+}: ContributorsChartProps) {
+  const dailyData = useMemo(() => getDailyContributorHistogramToDays({ stats, range }), [stats, range]);
```
Comment (nickytonline) [inline components/Graphs/ContributorsChart.tsx:29]: useMemo 🔥

**Review by nickytonline — APPROVED** [review]
Looks and works great @zeucapua!

:shipit:

**Review by bdougie — APPROVED** [review]

**Pull request merged**